pkg/internal/server/grpc/common_test.go

```diff
@@ -204,62 +234,176 @@ func (th *testHarness) makeFakeSnapshotClient() *fakesnapshot.Clientset {
 		ga := action.(clientgotesting.GetAction)
-		vs := volumeSnapshot(ga.GetName(), ga.GetNamespace())
+		vs := th.VolumeSnapshot(ga.GetName(), ga.GetNamespace())
 		return true, vs, nil
 	})
 	fakeSnapshotClient.PrependReactor("get", "volumesnapshotcontents", func(action clientgotesting.Action) (handled bool, ret apiruntime.Object, err error) {
 		ga := action.(clientgotesting.GetAction)
-		vsc := volumeSnapshotContent(ga.GetName(), th.DriverName)
+		vsc := th.VolumeSnapshotContent(ga.GetName(), th.DriverName)
 		return true, vsc, nil
 	})
+	fakeSnapshotClient.PrependReactor("get", "volumesnapshotclasses", func(action clientgotesting.Action) (handled bool, ret apiruntime.Object, err error) {
+		ga := action.(clientgotesting.GetAction)
+		if ga.GetName() != th.VolumeSnapshotClassName {
+			return false, nil, nil
+		}
+		vs := th.VSCNoAnn() // Not the default but parameters set
+		vs.Name = ga.GetName()
+		return true, vs, nil
+	})
 	return fakeSnapshotClient
 }
 
-func volumeSnapshot(name, namespace string) *snapshotv1.VolumeSnapshot {
+// FakeCountReactors saves the length of the fake reaction chains.
+// Implicitly called by WithFakeClientAPIs but can be explicitly
+// called if additional "permanent" reactors are added.
+func (th *testHarness) FakeCountReactors() {
+	th.lenFakeKubeClientReactionChain = len(th.FakeKubeClient.ReactionChain)
+	th.lenFakeSnapshotClientReactionChain = len(th.FakeSnapshotClient.ReactionChain)
+	th.lenFakeCBTClientReactionChain = len(th.FakeCBTClient.ReactionChain)
+}
+
+// FakePopPrependedReactors pops reactors from the front of the fake reaction chains.
+// Do not use if AddReactor was used after the call to CountFakeReactors.
+func (th *testHarness) FakePopPrependedReactors() {
+	var numToPop int
+
+	numToPop = len(th.FakeKubeClient.ReactionChain) - th.lenFakeKubeClientReactionChain
+	if numToPop > 0 {
+		th.FakeKubeClient.ReactionChain = th.FakeKubeClient.ReactionChain[numToPop:]
+	}
+
+	numToPop = len(th.FakeSnapshotClient.ReactionChain) - th.lenFakeSnapshotClientReactionChain
+	if numToPop > 0 {
+		th.FakeSnapshotClient.ReactionChain = th.FakeSnapshotClient.ReactionChain[numToPop:]
+	}
+
+	numToPop = len(th.FakeCBTClient.ReactionChain) - th.lenFakeCBTClientReactionChain
+	if numToPop > 0 {
+		th.FakeCBTClient.ReactionChain = th.FakeCBTClient.ReactionChain[numToPop:]
+	}
+}
+
+// ContentNameFromSnapshot returns the content name for a snapshot name.
+func (th *testHarness) ContentNameFromSnapshot(snapshotName string) string {
+	return "snapcontent-" + snapshotName
+}
+
+// SnapshotNameFromContent is the inverse of ContentNameFromSnapshot().
+func (th *testHarness) SnapshotNameFromContent(contentName string) string {
+	return strings.TrimPrefix(contentName, "snapcontent-") // Identity if prefix missing.
+}
+
+// HandleFromSnapshot returns a snapshot identifier based on the snapshot name.
+func (th *testHarness) HandleFromSnapshot(snapshotName string) string {
+	return snapshotName + "-id"
+}
```

Comment:
I wanted all objects to have distinct names to test templating. So now a VolumeSnapshotContent name is not equal to its VolumeSnapshot name but is a function of that.  This required changes in the handler tests, but not significant.